src/core/Akka/Actor/ActorCell.cs

```diff
@@ -77,90 +74,113 @@ private LatestFirstSystemMessageList UnstashAll()
+        /// is remotely deployed from elsewhere.
         /// </summary>
         public bool IsLocal { get { return true; } }
+        
         /// <summary>
-        /// TBD
+        /// A reference to the current actor instance
         /// </summary>
-        internal ActorBase Actor { get { return _actor; } }
+        internal ActorBase? Actor { get; private set; }
+
         /// <summary>
-        /// TBD
+        /// Indicates whether the actor is currently terminated.
         /// </summary>
-        public bool IsTerminated => Mailbox.IsClosed();
+        public bool IsTerminated => Mailbox!.IsClosed();
+        
         /// <summary>
-        /// TBD
+        /// A static reference to the current actor cell.
         /// </summary>
-        internal static ActorCell Current
+        internal static ActorCell? Current
         {
             get { return InternalCurrentActorCellKeeper.Current; }
         }
 
         /// <summary>
-        /// TBD
+        /// The <see cref="ActorSystem"/> this actor belongs to.
         /// </summary>
-        public ActorSystem System { get { return _systemImpl; } }
+        public ActorSystem System { get { return SystemImpl; } }
+        
         /// <summary>
-        /// TBD
+        /// INTERNAL API
         /// </summary>
-        public ActorSystemImpl SystemImpl { get { return _systemImpl; } }
+        public ActorSystemImpl SystemImpl { get; }
+
         /// <summary>
-        /// TBD
+        /// The <see cref="Props"/> used to create this actor.
         /// </summary>
-        public Props Props { get { return _props; } }
+        /// <remarks>
+        /// Will be re-used in the event that the actor restarts.
+        /// </remarks>
+        public Props Props { get; private set; }
+
         /// <summary>
-        /// TBD
+        /// The <see cref="IActorRef"/> instance that represents the current actor.
         /// </summary>
         public IActorRef Self { get { return _self; } }
         IActorRef IActorContext.Parent { get { return Parent; } }
+        
         /// <summary>
-        /// TBD
+        /// This actor's parent actor.
         /// </summary>
         public IInternalActorRef Parent { get; private set; }
+        
         /// <summary>
-        /// TBD
+        /// The actor who sent us <see cref="CurrentMessage"/>.
         /// </summary>
-        public IActorRef Sender { get; private set; }
+        /// <remarks>
+        /// Will be <c>null</c> when we are not processing messages.
+        /// </remarks>
+        public IActorRef? Sender { get; private set; }
+        
         /// <summary>
-        /// TBD
+        /// Will return <c>true</c> if <see cref="NumberOfMessages"/> is greater than zero.
         /// </summary>
-        public bool HasMessages { get { return Mailbox.HasMessages; } }
+        public bool HasMessages { get { return Mailbox!.HasMessages; } }
```

Comment:
`Mailbox` can't be `null` by the time anyone calls this method - or else there's been a catastrophic failure.